pkg/daemon/daemon.go

```diff
@@ -143,11 +143,11 @@ func (dn *Daemon) triggerUpdate() error {
 	return dn.update(currentConfig, desiredConfig)
 }
 
-// validate confirms that the node is actually in the state that it wants to be
-// in. it does this by looking at the elements in the target config and checks
-// if all are present on the node. if any file/unit is missing or there is a
-// mismatch, it re-triggers the update.
-func (dn *Daemon) validate() (bool, error) {
+// currentMatchesDesired confirms that the node is actually in the state that it
+// wants to be in. It does this by looking at the elements in the target config
+// and checks if all are present on the node. Returns true iif there are no
+// mismatches (e.g. files, units... XXX: but not yet OS version).
+func (dn *Daemon) currentMatchesDesired() (bool, error) {
```

Comment:
as far as bikeshedding this function name goes, I'm partial to keeping it `validate`, but if you think that's not descriptive enough, I think I would go with something like `isDesiredMachineState` or something like that. `currentMatchesDesired` says "helper function for checking if the current annotation equals the desired annotation" to me.

on that note, I notice now that the validate function actually does exactly that, and skips the rest of the validation if they are equal, but I really don't think it should. I think we should be validating the machine state every time we start up, whether or not the annotations imply that we've successfully validated it in the past. we can deal with that later though.

Reply:
I'll defer to @sdemos and @jlebon. I'm OK with either.